docs/source/routing/self-hosted/containerization/gcp.mdx

```diff
@@ -0,0 +1,161 @@
+---
+title: Deploying GraphOS Router on Google Cloud Platform
+subtitle: Deploy router with Google Cloud Run
+description: Build and deploy Apollo GraphOS Router or Apollo Router Core to Google Cloud Platform (GCP) with Google Cloud Run.
+---
+
+Learn how to deploy the router for development on Google Cloud Platform (GCP) with Google Cloud Run.
+
+You will:
+- Build a router image using a Dockerfile and a router configuration file.
+- Set up a container registry and push your router image to it.
+- Create a Cloud Run service and configure it to deploy your router.
+
+<Note>
+
+Not deploying to GCP? Try our router deployment guides for [AWS](/graphos/routing/self-hosted/containerization/aws) or [Azure](/graphos/routing/self-hosted/containerization/azure).
+
+</Note>
+
+## Prerequisites
+
+Before you start:
+
+1. [Set up a GraphQL API in GraphOS](/graphos/get-started/guides/graphql#step-1-set-up-your-graphql-api).
+    - Save your `APOLLO_KEY` and `APOLLO_GRAPH_REF` in your GCP Secret Manager. You'll need them when deploying the router.
+1. Install [Docker](https://www.docker.com/get-started/) locally.
+1. Create a [GCP account](https://cloud.google.com/) or use an existing account.
+1. Create a [GCP project](https://console.cloud.google.com/projectcreate). Choose a project name (for example, `my-project`) and save it to use later when deploying the router.
+1. Install the [gcloud CLI](https://cloud.google.com/sdk/docs/install) and log in to your GCP account.
+1. Choose a version of the router to deploy (for example, `v1.61.0`). You'll need it when specifying the router image to deploy.
+
+## Build router image
+
+To deploy your own router, start by customizing and building a router image, using a Dockerfile and a router configuration file:
+
+1. In a local directory, create a `router.yaml` file and copy-paste the following configuration into the file: 
```

Comment:
How does this `router.yaml` file relate to any existing router configuration? Should they also add their configurations here?